sdk/python/foundation-models/system/finetune/Llama-notebooks/text-generation/text-generation-samsum.ipynb

```diff
@@ -0,0 +1,794 @@
+   "metadata": {},
+   "source": [
+    "Define finetune parameters\n",
+    "\n",
+    "Finetune parameters can be grouped into 2 categories - training parameters, optimization parameters\n",
+    "\n",
+    "Training parameters define the training aspects such as - \n",
+    "1. the optimizer, scheduler to use\n",
+    "2. the metric to optimize the finetune\n",
+    "3. number of training steps and the batch size\n",
+    "and so on\n",
+    "\n",
+    "Optimization parameters help in optimizing the GPU memory and effectively using the compute resources. Below are few of the parameters that belong to this category. _The optimization parameters differs for each model and are packaged with the model to handle these variations._\n",
+    "1. enable the deepspeed, ORT and LoRA\n",
+    "2. enable mixed precision training\n",
+    "2. enable multi-node training "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Training parameters\n",
+    "training_parameters = dict(\n",
+    "    num_train_epochs=3,\n",
+    "    per_device_train_batch_size=1,\n",
+    "    per_device_eval_batch_size=1,\n",
+    "    learning_rate=2e-5,\n",
+    ")\n",
+    "print(f\"The following training parameters are enabled - {training_parameters}\")\n",
+    "\n",
+    "# Optimization parameters - As these parameters are packaged with the model itself, lets retrieve those parameters\n",
+    "if \"model_specific_defaults\" in foundation_model.tags:\n",
+    "    optimization_parameters = ast.literal_eval(\n",
+    "        foundation_model.tags[\"model_specific_defaults\"]\n",
+    "    )  # convert string to python dict\n",
+    "else:\n",
+    "    optimization_parameters = dict(\n",
+    "        apply_lora=\"true\", apply_deepspeed=\"true\", apply_ort=\"true\"\n",
+    "    )\n",
+    "print(f\"The following optimizations are enabled - {optimization_parameters}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.dsl import pipeline\n",
+    "from azure.ai.ml.entities import CommandComponent, PipelineComponent, Job, Component\n",
+    "from azure.ai.ml import PyTorchDistribution, Input\n",
+    "\n",
+    "# fetch the pipeline component\n",
+    "pipeline_component_func = registry_ml_client.components.get(\n",
+    "    name=\"text_generation_pipeline\", label=\"latest\"\n",
+    ")\n",
+    "\n",
+    "\n",
+    "# define the pipeline job\n",
+    "@pipeline()\n",
+    "def create_pipeline():\n",
+    "    text_classification_pipeline = pipeline_component_func(\n",
+    "        # specify the foundation model available in the azureml system registry id identified in step #3\n",
+    "        mlflow_model_path=foundation_model.id,\n",
+    "        # huggingface_id = 'bert-base-uncased', # if you want to use a huggingface model, uncomment this line and comment the above line\n",
```

Comment:
shall we replace this _huggingface_id_ with llama's huggingface id?

Reply:
Changed anyway